crates/core/src/rpc/minimal.rs

```diff
@@ -586,42 +588,28 @@ impl Minimal for SurfpoolMinimalRpc {
             Err(e) => return Box::pin(future::err(e)),
         };
 
-        let state_reader = match meta.get_state() {
+        let svm_locker = match meta.get_svm_locker() {
             Ok(res) => res,
             Err(e) => return Box::pin(future::err(e.into())),
         };
-        let account = state_reader.svm.get_account(&pubkey);
-
-        // Drop the lock on the state while we fetch accounts
-        let absolute_slot = state_reader.epoch_info.absolute_slot;
-        let rpc_client = RpcClient::new(state_reader.rpc_url.clone());
-        drop(state_reader);
 
         Box::pin(async move {
-            let account = if let None = account {
-                // Fetch and save the missing account
-                if let Some(fetched_account) = rpc_client.get_account(&pubkey).await.ok() {
-                    let mut state_reader = meta.get_state_mut()?;
-                    state_reader
-                        .svm
-                        .set_account(pubkey, fetched_account.clone())
-                        .map_err(|err| {
-                            Error::invalid_params(format!(
-                                "failed to save fetched account {pubkey:?}: {err:?}"
-                            ))
-                        })?;
-
-                    Some(fetched_account)
-                } else {
-                    None
-                }
-            } else {
-                account
+            let mut svm_writer = svm_locker.write().await;
+            let res = svm_writer
+                .get_account_mut(
+                    &pubkey,
+                    GetAccountStrategy::LocalThenConnectionOrDefault(None),
+                )
+                .await
+                .unwrap();
+            let balance = match res {
+                Some(account) => account.lamports,
+                None => 0,
```

Comment:
I think this should return an `AccountNotFound` error, not 0

Reply:
Are you sure?

Reply:
Oooo, perhaps not. The solana cli returns an error:
```sh
 ~ solana account zbBjhHwuqyKMmz8ber5oUtJJ3ZV4B6ePmANfGyKzVGX --url devnet
Error: AccountNotFound: pubkey=zbBjhHwuqyKMmz8ber5oUtJJ3ZV4B6ePmANfGyKzVGX
```

but the `getBalance` rpc method when I hit devnet returns 0. Good call pushing back 🙏